crates/extensions/tedge_flows/src/flow.rs

```diff
@@ -0,0 +1,269 @@
+        Ok(messages)
+    }
+
+    pub async fn on_interval(
+        &mut self,
+        js_runtime: &JsRuntime,
+        stats: &mut Counter,
+        timestamp: &DateTime,
+    ) -> Result<Vec<Message>, FlowError> {
+        let stated_at = stats.flow_on_interval_start(self.source.as_str());
+        let mut messages = vec![];
+        for step in self.steps.iter() {
+            let js = step.script.source();
+            // Process first the messages triggered upstream by the tick
+            let mut transformed_messages = vec![];
+            for message in messages.iter() {
+                let step_started_at = stats.flow_step_start(&js, "onMessage");
+                let step_output = step.script.on_message(js_runtime, timestamp, message).await;
+                match &step_output {
+                    Ok(messages) => {
+                        stats.flow_step_done(&js, "onMessage", step_started_at, messages.len())
+                    }
+                    Err(_) => stats.flow_step_failed(&js, "onMessage"),
+                }
+                transformed_messages.extend(step_output?);
+            }
+
+            // Only then process the tick
+            let step_started_at = stats.flow_step_start(&js, "onInterval");
+            let tick_output = step.script.on_interval(js_runtime, timestamp).await;
+            match &tick_output {
+                Ok(messages) => {
+                    stats.flow_step_done(&js, "onInterval", step_started_at, messages.len())
+                }
+                Err(_) => stats.flow_step_failed(&js, "onInterval"),
+            }
+            transformed_messages.extend(tick_output?);
+
+            // Iterate with all the messages collected at this step
+            messages = transformed_messages;
+        }
+        stats.flow_on_interval_done(self.source.as_str(), stated_at, messages.len());
+        Ok(messages)
+    }
+}
+
+impl FlowStep {
+    pub(crate) fn check(&self, flow: &Utf8Path) {
+        let script = &self.script;
+        if script.no_js_on_message_fun {
+            warn!(target: "flows", "Flow script with no 'onMessage' function: {}", script.path.display());
+        }
+        if script.no_js_on_config_update_fun && !self.config_topics.is_empty() {
+            warn!(target: "flows", "Flow script with no 'onConfigUpdate' function: {}; but configured with 'config_topics' in {flow}", script.path.display());
+        }
+        if script.no_js_on_interval_fun && script.interval_secs != 0 {
+            warn!(target: "flows", "Flow script with no 'onInterval' function: {}; but configured with an 'interval' in {flow}", script.path.display());
+        }
+    }
+
+    pub(crate) fn fix(&mut self) {
+        let script = &mut self.script;
+        if !script.no_js_on_interval_fun && script.interval_secs == 0 {
+            // 0 as a default is not appropriate for a script with an onInterval handler
+            script.interval_secs = 1;
```

Comment:
Why **silently** fix a user misconfiguration? Wouldn't it be better to just log this as misconfiguration in the `check() itself, and avoid loading/reloading the contained flow?